pkg/destroy/ibmcloud/ibmcloud.go

```diff
@@ -74,9 +76,41 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 		UserProvidedSubnets: metadata.ClusterPlatformMetadata.IBMCloud.Subnets,
 		UserProvidedVPC:     metadata.ClusterPlatformMetadata.IBMCloud.VPC,
 		pendingItemTracker:  newPendingItemTracker(),
+		maxRetryAttempt:     10,
+		maxRetryGap:         40,
 	}, nil
 }
 
+// Retry ...
+func (o *ClusterUninstaller) Retry(funcToRetry func() (error, bool)) error {
+	var err error
+	var stopRetry bool
+	retryGap := 10
+	for i := 0; i < o.maxRetryAttempt; i++ {
+		if i > 0 {
+			time.Sleep(time.Duration(retryGap) * time.Second)
+		}
+		// Call function which required retry, retry is decided by function itself
+		err, stopRetry = funcToRetry()
+		if stopRetry {
+			break
+		}
+
+		// Update retry gap as per exponentioal
```

Comment:
I did follow the library flow we do it exponential for delete volume wait but I have no issue in doing it with constant retry , I am removing that and put max attempts as 30 so total 5 mins we will wait for volume to delete else it will be retried by the existing installer framework using the pendingdiskitems logic.